src/pipeline/common/PullRequest.groovy

```diff
@@ -0,0 +1,61 @@
+package pipeline.common
+
+def addCommentsToPR(String library_dir, String manifest_path, String public_jenkins_url){
+    try{
+        withEnv([
+            "BUILD_URL=${env.BUILD_URL}",
+            "JENKINS_URL=${env.JENKINS_URL}"
+        ]){
+            withCredentials([string(credentialsId: 'JENKINSRHD_GITHUB_TOKEN',
+                                    variable: 'GITHUB_TOKEN')])
+            {
+                sh """#!/bin/bash 
+                ./$library_dir/build-release-tools/HWIMO-BUILD ./$library_dir/build-release-tools/application/post-result.py \
+                --manifest_file ${manifest_path} \
+                --jenkins_url ${JENKINS_URL} \
+                --build_url ${BUILD_URL} \
+                --public_jenkins_url ${public_jenkins_url} \
+                --ghtoken ${GITHUB_TOKEN}
+                """
+            }
+        }
+    } catch(error){
+        echo "Caught error during adding comments to pull request: ${error}"
+    }
+}
+
+def setStatusOfPR(String library_dir, String manifest_path, String public_jenkins_url){
+    try{
+        // if previous steps all pass,  $currentBuild.result will be set to "SUCCESS" explictly in pipeline groovy code
+        // if Junit plugin found test case error in previous step,  the plugin will set $currentBuild.result  to "Unstable"
+        // if previous steps abort with error, the $currentBuild.result will not get chance to be set . so value is "null" here
+        // ------
+        //Jenkins currentBuild.result| github commit status(https://developer.github.com/v3/repos/statuses/ )
+        // null                      | failure
+        // failure                   | failure
+        // unstable                  | failure
+        // success                   | success
+        if ("${currentBuild.result}" != "SUCCESS"){
+            currentBuild.result = "FAILURE"
+        }
+        String build_status = currentBuild.result.toLowerCase()
+        withEnv([
+            "BUILD_URL=${env.BUILD_URL}"
```

Comment:
where does this parameter comes from ?
GHPRB build-in parameter ? if yes, please comment it in code ,otherwise, it cause confusion

Reply:
Thanks, fixed